packages/core/src/sync/workers/sync.ts

```diff
@@ -66,7 +67,7 @@ export class SyncWorker implements Worker<SyncJobData> {
    */
   async handler(job: PgBoss.Job) {
     const jobData = job.data as SyncJobData
-    const { fromBlock, toBlock, models } = jobData
+    const { fromBlock, toBlock, models, syncCompleteCallback } = jobData
```

Comment:
Interesting but unfortunately pg.boss serializes the data when its stored in postgres so syncCompleteCallback will not be a function but rather a stringified function. 

You could pass a callback to the worker itself